js/huobipro.js

```diff
@@ -1017,4 +1019,154 @@ module.exports = class huobipro extends Exchange {
+        //         'address': '1QR....',
+        //         'address-tag': '',
+        //         'fee': 0.0005,
+        //         'state': 'confirmed',
+        //         'created-at': 1552107295685,
+        //         'updated-at': 1552108032859
+        //     }
+        //
+        let timestamp = this.safeInteger (transaction, 'created-at');
+        if (timestamp !== undefined) {
+            timestamp = parseInt (timestamp);
+        }
+        let updated = this.safeInteger (transaction, 'updated-at');
+        if (updated !== undefined) {
+            updated = parseInt (updated);
+        }
+        let code = this.safeCurrencyCode (transaction, 'currency');
+        if (currency === undefined) {
+            currency = this.safeValue (this.currencies_by_id, code);
+        }
+        if (currency !== undefined) {
+            code = currency['code'];
+        } else {
+            code = this.commonCurrencyCode (code);
+        }
+        let type = this.safeString (transaction, 'type');
+        if (type === 'withdraw') {
+            type = 'withdrawal';
+        }
+        let status = this.parseTransactionStatusByType (this.safeString (transaction, 'state'), type);
+        let tag = this.safeString (transaction, 'address-tag');
+        let feeCost = this.safeFloat (transaction, 'fee');
+        if (feeCost !== undefined) {
+            feeCost = Math.abs (feeCost);
+        }
+        return {
+            'info': transaction,
+            'id': this.safeString (transaction, 'id'),
+            'txid': this.safeString (transaction, 'tx-hash'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'address': this.safeString (transaction, 'address'),
+            'tag': tag,
+            'type': type,
+            'amount': this.safeFloat (transaction, 'amount'),
+            'currency': code,
+            'status': status,
+            'updated': updated,
+            'fee': {
+                'currency': code,
+                'cost': feeCost,
+                'rate': undefined,
+            },
+        };
+    }
+
+    parseTransactionStatusByType (status, type = undefined) {
+        if (type === undefined) {
+            return status;
+        }
+        let statuses = {
+            'deposit': {
+                'unknown': 'pending',
+                'confirming': 'pending',
+                'confirmed': 'ok',
+                'safe': 'ok',
+            },
+            'withdrawal': {
+                'submitted': 'pending',
+                'canceled': 'canceled',
+                'reexamine': 'pending',
+                'reject': 'failed',
+                'pass': 'pending',
+                'wallet-reject': 'failed',
+                'confirmed': 'ok',
+                'confirm-error': 'failed',
+                'repealed': 'failed',
+                'wallet-transfer': 'pending',
+                'pre-transfer': 'pending',
+            },
+        };
```

Comment:
I have a few more statuses in my homegrown implementation of `parseTransactionStatus` some of them may be out of date but some of them are still mentioned in the [docs](https://github.com/huobiapi/API_Docs_en/wiki/REST_Reference#withdraw-states). So could you please check it one more time:

```javascript
const statuses = {
        // deposit statuses
        'unknown': 'failed',
        'confirming': 'pending',
        'confirmed': 'ok',
        'safe': 'ok',
        'orphan': 'failed',
        // withdrawal statuses
        'submitted': 'pending',
        'reexamine': 'pending',
        'canceled': 'canceled',
        'pass': 'ok',
        'reject': 'failed',
        'pre-transfer':	'pending',
        'wallet-transfer': 'pending',
        'wallet-reject': 'failed',
        // 'confirmed': 'ok', // already present among deposit statuses
        'confirm-error': 'failed',
        'repealed': 'canceled',
        'valid': 'ok',
    }
```

Reply:
Checked the docs and updated the statuses accordingly thanks